contact_manager.py

```diff
@@ -0,0 +1,218 @@
+import json
+
+
+class Address:
+
+    def __init__(self, street, city, state, pin_code):
+        self.street = street
+        self.city = city
+        self.state = state
+        self.pin_code = pin_code
+
+
+class Contact:
+
+    def __init__(self, name, phone_no, email, street, city, state, pin_code):
+        self.name = name
+        self.phone_no = phone_no
+        self.email = email
+        self.address = Address(street, city, state, pin_code)
+
+
+class Contacts:
+
+    provider = {'Airtel': ['9900', '9800', '9811'], 'BSNL': ['9440', '9822'], 'Idea': ['9848', '9912'],
+                'Reliance': ['9300', '9812']}
+
+    def __init__(self, data):
+        self.data = data
+
+    def add_person(self, contact):
+        self.data[contact.phone_no] = contact
+
+    def modify_person(self, phone_no, fields):
+        for key in fields.keys():
+            if key == 'name':
+                self.data[phone_no].name = fields[key]
+            elif key == 'phone_no':
+                self.data[phone_no].phone_no = fields[key]
+                self.data[fields[key]] = self.data.pop[phone_no]
+            elif key == 'email':
+                self.data[phone_no].email = fields[key]
+            elif key == 'street':
+                self.data[phone_no].addres.street = fields[key]
+            elif key == 'city':
+                self.data[phone_no].addres.city = fields[key]
+            elif key == 'state':
+                self.data[phone_no].addres.state = fields[key]
+            elif key == 'pin_code':
+                    self.data[phone_no].addres.pin_code = fields[key]
+
+    def delete_person(self, phone_no):
+        del self.data[phone_no]
+
+    def get_details(self, phone_no):
+        return self.data[phone_no]
+
+    def get_provider(self, phone_no):
```

Comment:
Create a provider Class!! Provide Manager!! Initialize all provider and ProviderManager!! Checking for aprovider should happen in ProviderManager!!